app/main.py

```diff
@@ -1,10 +1,21 @@
 from fastapi import FastAPI, HTTPException, Query
+from fastapi.responses import HTMLResponse
+from fastapi.staticfiles import StaticFiles
 from pydantic import HttpUrl
 import requests
 from bs4 import BeautifulSoup
+import os
 
 app = FastAPI()
 
+# Mount the static files directory
+app.mount("/static", StaticFiles(directory="templates/static"), name="static")
+
+@app.get("/", response_class=HTMLResponse)
+def read_root():
+    with open(os.path.join("templates", "index.html")) as f:
```

Comment:
Is this the best way to serve this html?

Reply:
Thank you for the feedback! I understand the concern. Serving HTML files directly from the filesystem is not the most efficient or secure way. Instead, we can use FastAPI's `Jinja2Templates` to serve HTML templates.

I'll make the necessary changes to use `Jinja2Templates` for serving the HTML file and update the PR.